src/Filter/Validator/ValidatorInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Filter\Validator;
+
+use Symfony\Component\HttpFoundation\Request;
+
+interface ValidatorInterface
+{
+    public function validate(string $name, array $filterDescription, Request $request): array;
```

Comment:
I think we should use a context instead of depending of the HttpFoundation Request, like the Doctrine filters. It will more independent / maintainable this way. WDYT?

Reply:
Good catch @alanpoulain. We try to remove the hard dependency to HttpFoundation to be able to support PSR-7 at some point. The way to go is using the context.

Reply:
As the listener is a Symfony request listener, do you think I should only call `$request->query->all()` ?

Reply:
You should have a look at the https://github.com/api-platform/core/blob/master/src/EventListener/ReadListener.php.